### PR TITLE
Finish remainder of cli conversion

### LINKED ISSUE #1154: Use cli for feedback to user

### LINKED ISSUE #1207: `write_if_different()` prints rather than a message
Is it intentional that `write_if_different()` uses `cat()` instead of `inform()` or some other type of message? I noticed that `roxygenise()` requires both `capture.output()` and `suppressMessages()` to shush.

### PR DESCRIPTION
**Pull request by hadley**
Fixes #1154. Fixes #1207.

### CODE REVIEW

File: R/load.R [hunk 1]
```diff
@@ -82,18 +82,18 @@ find_load_strategy <- function(x, option = roxy_meta_get("load", "pkgload")) {
   if (is.null(x)) {
     x <- option
     if (!is.character(x) || length(x) != 1) {
-      abort("roxygen2 `load` option must be a string")
+      cli::cli_abort("roxygen2 `load` option must be a string")
```
Comment (gaborcsardi) [inline R/load.R:85]: To we want to write 
```
      cli::cli_abort("roxygen2 {.code load} option must be a string")
```
here and also below?  At some point we should write up some conventions.

File: R/rd-template.R [hunk 2]
```diff
@@ -42,7 +42,9 @@ template_find <- function(base_path, template_name) {
   path_exists <- file.exists(path)
 
   if (!any(path_exists)) {
-    stop("Can't find template '", template_name, "'", call. = FALSE)
+    # This should really use warn_roxy_tag() but it's not worth refactoring
+    # for this rarely used feature
+    cli::cli_abort("Can't find template {.str {template_name}}'", call = NULL)
```
Comment (gaborcsardi) [inline R/rd-template.R:47]: Spurious single quote at the end?

**Review by gaborcsardi — APPROVED** [review]
Two small comments.

**Comment by gaborcsardi** [comment]
I have no clue what's up with the windows CI. It is unlikely that both windows jobs have a corrupt cache.

File: R/load.R (hunk 1 above)
Comment (hadley) [reply R/load.R:85]: Or `.arg`?

**Comment by hadley** [comment]
@gaborcsardi hmmm, I polished that test and I guess I must've messed it up?

**Comment by hadley** [comment]
@gaborcsardi ooops yeah, I broke the skip — the windows GitHub runner must have `autocrlf = true` which the skip was supposed to avoid.

**Pull request merged**